eth/tracers/internal/tracetest/supply_test.go

```diff
@@ -0,0 +1,577 @@
+// Copyright 2021 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package tracetest
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"os"
+	"path"
+	"path/filepath"
+	"reflect"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/consensus/beacon"
+	"github.com/ethereum/go-ethereum/consensus/ethash"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/eth/tracers"
+	"github.com/ethereum/go-ethereum/params"
+
+	// Force-load live packages, to trigger registration
+	_ "github.com/ethereum/go-ethereum/eth/tracers/live"
+)
+
+type supplyInfo struct {
+	Delta       *big.Int `json:"delta"`
+	Reward      *big.Int `json:"reward"`
+	Withdrawals *big.Int `json:"withdrawals"`
+	Burn        *big.Int `json:"burn"`
```

Comment:
`big.Int` marshals as json numbers. Which means that we're kind of limited to `53` bits, because of javascript. So better if we marshal it as hex-strings (or dec-strings for that matter).

Reply:
So Delta is a signed number. The best thing I can think of is to add a new field for the sign.

Reply:
I like your idea @s1na, as supporting Negative numbers for hex isn't the nicest thing and also the consumers of the JSON will have to find a library that supports negative numbers as well.

Another suggestion might be if we use two different fields like `DeltaInflation` and `DeltaDeflation` (we have to think of better names) (thanks to @meowsbits for this suggestion)

Reply:
What do you think of this? https://github.com/ziogaschr/go-ethereum/commit/3ac74bd0ed6b02331128c7f23f28141d9ec7d499#diff-220c3808fee4697f80eb786496b8d85fc0c3f9a215b9d09e13bdecc6b87a82f0R55

Reply:
@holiman @s1na I pushed some commits which change the output to the following after our discussion with @s1na.
We dropped the `delta`, which could have negative numbers and we will let the consumer to do the calculations on their side.

```json
{
  "issuance": {
    "genesisAlloc": "0x0", // omitempty
    "reward": "0x0",
    "withdrawals": "0x0"
  },
  "burn": {
    "1559": "0x0",
    "blob": "0x0",
    "misc": "0x0"
  },
  "blockNumber": 1,
  "hash": "0x54177b2adb754306d0d267f82537551c467201bf0c8509e60b77229b41c99e3a",
  "parentHash": "0xaf41e72f748de317965454508c749f7e14dc4fe444cd07bca4c981c7e952364d"
}
```